dds/DCPS/transport/rtps_udp/RtpsUdpTransport.cpp

```diff
@@ -427,6 +427,13 @@ RtpsUdpTransport::update_locators(const RepoId& remote,
   }
 }
 
+void
+RtpsUdpTransport::get_and_reset_relay_message_counts(RelayMessageCounts& counts)
+{
+  counts = relay_message_counts_;
```

Comment:
I know it's not on master yet, but I'm pretty sure this is going to eventually cause trouble with ThreadSanitizer tests without some kind of synchronization between threads, assuming whatever thread is querying the stats is not the same as the write thread and/or the transport reactor thread. That said, I don't see anything in this PR actually making the call into get_and_reset (presumably happens in the relay?) so maybe not an issue yet.